framework/packages/abstract-app/src/interface.rs

```diff
@@ -41,134 +41,166 @@
+                env: ::cosmwasm_std::Env,
+                msg: ::cosmwasm_std::Reply,
+            ) -> Result<
+                ::cosmwasm_std::Response,
+                <$app_type as $crate::abstract_sdk::base::Handler>::Error,
+            > {
+                use $crate::abstract_sdk::base::ReplyEndpoint;
+                $app_const.reply(deps, env, msg)
+            }
 
-	        pub type InstantiateMsg = <$app_type as ::abstract_sdk::base::InstantiateEndpoint>::InstantiateMsg;
-	        pub type ExecuteMsg = <$app_type as ::abstract_sdk::base::ExecuteEndpoint>::ExecuteMsg;
-	        pub type QueryMsg = <$app_type as ::abstract_sdk::base::QueryEndpoint>::QueryMsg;
-	        pub type MigrateMsg = <$app_type as ::abstract_sdk::base::MigrateEndpoint>::MigrateMsg;
-	    }
+            pub fn sudo(
+                deps: ::cosmwasm_std::DepsMut,
+                env: ::cosmwasm_std::Env,
+                msg: <$app_type as $crate::abstract_sdk::base::Handler>::SudoMsg,
+            ) -> Result<
+                ::cosmwasm_std::Response,
+                <$app_type as $crate::abstract_sdk::base::Handler>::Error,
+            > {
+                use $crate::abstract_sdk::base::SudoEndpoint;
+                $app_const.sudo(deps, env, msg)
+            }
 
-	    pub mod interface{
+            pub type InstantiateMsg =
+                <$app_type as $crate::abstract_sdk::base::InstantiateEndpoint>::InstantiateMsg;
+            pub type ExecuteMsg =
+                <$app_type as $crate::abstract_sdk::base::ExecuteEndpoint>::ExecuteMsg;
+            pub type QueryMsg = <$app_type as $crate::abstract_sdk::base::QueryEndpoint>::QueryMsg;
+            pub type MigrateMsg =
+                <$app_type as $crate::abstract_sdk::base::MigrateEndpoint>::MigrateMsg;
+        }
+
+        pub mod interface {
+            use super::_wrapper_fns;
             use super::*;
-	    	use super::_wrapper_fns;
-	    	#[::cw_orch::interface(_wrapper_fns::InstantiateMsg, _wrapper_fns::ExecuteMsg, _wrapper_fns::QueryMsg, _wrapper_fns::MigrateMsg)]
-			pub struct $interface_name;
-
-
-			impl <Chain: ::cw_orch::prelude::CwEnv> ::cw_orch::prelude::Uploadable for $interface_name<Chain> {
-			    fn wasm(&self) -> ::cw_orch::prelude::WasmPath {
-			    	let wasm_name = env!("CARGO_CRATE_NAME").replace('-', "_");
-			        ::cw_orch::prelude::ArtifactsDir::auto(Some(env!("CARGO_MANIFEST_DIR").to_string()))
-			        	.find_wasm_path(&wasm_name).unwrap()
-			    }
+            #[::cw_orch::interface(
+                _wrapper_fns::InstantiateMsg,
+                _wrapper_fns::ExecuteMsg,
+                _wrapper_fns::QueryMsg,
+                _wrapper_fns::MigrateMsg
+            )]
+            pub struct $interface_name;
+
+            impl<Chain: ::cw_orch::prelude::CwEnv> ::cw_orch::prelude::Uploadable
+                for $interface_name<Chain>
+            {
+                fn wasm(&self) -> ::cw_orch::prelude::WasmPath {
```

Comment:
I'm not entirely sure if we should re-expose cw-orch as well. I have a gut feeling that it could cause some weird versioning issues when users try to use it with other packages that also use cw-orch, but could be wrong.

Reply:
Here's what chat-gpt thought: 

Re-exporting a crate in Rust can be a convenient feature, allowing users of your library to access types or functions from a dependency without having to add that dependency themselves. However, there are potential downsides and complexities, especially concerning versioning and compatibility.

### Downsides of Re-exporting a Crate

1. **Version Lock-In**: When you re-export a crate, you lock your users into the specific version of that crate you're using. If they need features or fixes from a newer version (or need to use an older version for compatibility reasons), they might face conflicts.

2. **Dependency Transparency**: Re-exporting hides the dependency from the users of your library. While this can be seen as an advantage (simplifying their `Cargo.toml`), it also means they may not be aware of the underlying dependencies, which could be relevant for understanding behavior, debugging, or auditing dependencies for security reasons.

3. **Semver Compatibility**: If you make a significant change to the version of a re-exported crate (like updating from 2.0 to 3.0), it could be a breaking change for your library users. This requires careful management of your own library's versioning to adhere to semantic versioning principles.

4. **Increased Compile Time**: Re-exporting a crate means it's always compiled when your library is compiled, even if the end user doesn't need that particular functionality. This can lead to increased compilation times.

5. **Namespace Pollution**: Re-exporting everything from a crate can pollute the namespace of your library, potentially leading to confusion or naming conflicts.

6. **Feature Control**: When using features of a dependency, you need to ensure that the features are enabled or disabled appropriately in your crate. This can become complex if the features of the re-exported crate need to be configurable by the end user.

### Mitigation Strategies

- **Versioning Carefully**: Choose a widely compatible version to re-export, and be cautious about updating it. Consider the impact of any version change on your users.
  
- **Documentation**: Clearly document the fact that you're re-exporting a crate, including the version and any relevant information about the dependency.

- **Selective Re-exporting**: Instead of re-exporting the entire crate, only re-export the parts that are necessary for your library's API. This reduces namespace pollution and makes it clearer what parts of the external crate are actually being used.

- **Feature Flags**: Use feature flags in your crate to allow users to opt-in or opt-out of certain dependencies, including re-exported crates.

### Conclusion

While re-exporting can make libraries more user-friendly and reduce boilerplate for end users, it requires thoughtful consideration regarding versioning, dependency management, and public API design. It's essential to balance convenience with these potential complexities to ensure a smooth experience for your library's users.

Reply:
Feature-flag wise it would mean re-exposing all the cw-orch flags in the app, which IMO is weird. 
For the other packages it makes sense because core and sdk are both also part of Abstract. 

While cw-orch is a completely different repo/package

Reply:
> I'm not entirely sure if we should re-expose cw-orch as well. I have a gut feeling that it could cause some weird versioning issues when users try to use it with other packages that also use cw-orch, but could be wrong.

Yeah, also not 100% not sure about it, need more testing. Let me know if you get error "similar name but two distinct types" thing